R/cie.R

```diff
@@ -64,15 +84,41 @@ cie <- function(vismodeldata, space = c("XYZ", "LAB", "LCh")) {
 
     # Calculate tristimulus values for neutral point. First need to
     # re-grab original sensitivity and illuminant data.
-    S <- attr(vismodeldata, "data.visualsystem.chromatic")
-    illum <- attr(vismodeldata, "data.illuminant") # Illuminant
+    if ("vismodel" %in% class(vismodeldata)) {
+      S <- attr(vismodeldata, "data.visualsystem.chromatic")
+      illum <- attr(vismodeldata, "data.illuminant") # Illuminant
+    } else {
+      # Grab built-in data
+      sens <- vissyst
+      bgil <- bgandilum
+
+      # Match user-specified arguments
+      visual2 <- tryCatch(
+        match.arg(visual),
+        error = function(e) "user-defined"
```

Comment:
Sure, I've sketched one into `cie()` as an idea. Now I think about it a bit more though, I'm not actually sure. If people are specifying their own illuminant/vis, do they need to be told they're doing so? I guess I'm a little wary of adding superfluous messages, which can be annoying.

I'll leave it in `cie()` since it's new, but might hold off on `vismodel()` for the moment. Thoughts always welcome.